src/server.js

```diff
@@ -0,0 +1,150 @@
+"use strict";
+var http = require('http');
+var graphql = require('graphql');
+require('babel-register');
+var WebSocketServer = require('websocket').server;
+var graphql_tools = require('graphql-tools');
+var graphql_validator = require('graphql/validation');
+var graphql_execution = require('graphql/execution');
+
+class Server {
+  /*
+  options {
+    schema: GraphQLSchema
+    contextValue?: any,
+    rootValue?: any,
+    formatResponse?: (Object) => Object,
+    validationRules?: Array<any> 
+    triggerGenerator?: (Object) => [Object]
+  }
+  */
+  constructor(options, httpServer) {
+    this.options = options;
+    this.triggers = {}; // trigger_object: [{connection, sub_id}]
+    //initialize http server
+
+    //init and connect websocket server to http
+    this.wsServer = new WebSocketServer({
+      httpServer: httpServer,
+      autoAcceptConnections: false
+    });
+
+    this.wsServer.on('request', (request) => {
+      //accept connection
+      var connection = request.accept('graphql-protocol', request.origin);
```

Comment:
Maybe provide a way for users to restrict origin?
And probably should only accept connection if it's asking for 'graphql-protocol'.
Not sure, but maybe protocol will also have a verison in the future.
Think about whether protocol name should be an option to the server?